receiver/podmanreceiver/podman_client.go

```diff
@@ -121,6 +134,33 @@ func (c *podmanClient) stats() ([]containerStats, error) {
 	}
 	return report.Stats, nil
 }
+func (c *podmanClient) events() (chan Event, error) {
+	ch := make(chan Event)
+	params := url.Values{}
+	params.Add("stream", "true")
+	params.Add("since", "0m")
+
+	response, err := c.request(context.Background(), "/events", params)
+	if err != nil {
+		return nil, err
+	}
+
+	go func() {
+		dec := json.NewDecoder(response.Body)
+		for {
+			var event Event
+			if err := dec.Decode(&event); err != nil {
+				if err == io.EOF {
+					break
+				}
+				log.Fatalf("error decoding event: %w", err)
```

Comment:
we should restart the watch in case of failure or stream shutdown probably with some exponential delay